### PR TITLE
Renamed repos to data provider

### LINKED ISSUE #12553: Reader Improvements: Repository Renaming
For the RI project the term repository doesn't quite fit the definition of a single point of entry for all data access. As such we will rename the repository to **dataProvider** and keep them lean and focused on get responsibilities only. 

The goals of this task are:
- [x] Rename `ReaderPostRepository` to `ReaderPostDataProvider`

- [x]  Rename `ReaderDiscoverRepository` to  `ReaderDiscoverDataProvider`. 

- [x] Remove the like action functionality

### PR DESCRIPTION
**Pull request by zwarm**
Fixes #12553 

- Renames `ReaderPostRepository` to `ReaderPostDataProvider`
- Renames `ReaderDiscoverRepository` to `ReaderDiscoverDataProvider.`
- Removes the like action functionality
- Adds `Started` to `ReaderRepositoryCommunication` and replaced Success in `FetchPostsForTagUseCase` and `FetchDiscoverCardsUseCase`

To test
Open the app with 
Set FF_READER_IMPROVEMENTS_PHASE_2 to true
Navigate to reader
Discover feed should load

PR submission checklist:

- [X] I have considered adding unit tests where possible.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModel.kt [hunk 1]
```diff
@@ -58,7 +58,7 @@ class ReaderDiscoverViewModel @Inject constructor(
     private val photonWidth: Int = 500
     private val photonHeight: Int = 500
 
-    private lateinit var readerDiscoverRepository: ReaderDiscoverRepository
+    private lateinit var mReaderDiscoverDataProvider: ReaderDiscoverDataProvider
```
Comment (jd-alexander) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModel.kt:61]: Hi @zwarm for kotlin files, I don't think the member` m` in front of the variable is used. Let me know your thoughts on this if there's another reason it was used. Thank you 🙏

Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModel.kt:61]: Yes, you are correct. That was AndroidStudio auto rename/refactor. Nice catch.

Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModel.kt]: Replaced in c39c765

**Comment by jd-alexander** [comment]
> Thanks @zwarm! LGTM ;)

Same here, thanks @zwarm 🥇 

> @jd-alexander I hope you won't mind me taking over the PR. I thought it'd be better to merge this PR asap in other to avoid unnecessary conflicts. ;)

I don't mind at all @malinajirka Thanks for the help!

**Pull request merged**